### PR TITLE
fix(identity/featurestates): fix permission for /all endpoint

### LINKED ISSUE #1455: Permission denied for users with Manage Identities permission when calling /all on identity feature states view
Calling `GET https://api.flagsmith.com/api/v1/environments/:key/identities/:id/featurestates/all/` when authenticated as a user with the MANAGE_IDENTITIES permission results in a 403 permission denied. Users with this permission should have access to this endpoint. 

We should also verify the same situation for `.../edge-identities/:uuid/featurestates/all/`.

### PR DESCRIPTION
**Pull request by gagantrivedi**

### CODE REVIEW

File: api/conftest.py [hunk 1]
```diff
@@ -170,3 +176,41 @@ def master_api_key_client(master_api_key):
...
     return api_client
+
+
+@pytest.fixture()
+def test_user(django_user_model):
+    return django_user_model.objects.create(email="user@example.com")
+
+
+@pytest.fixture()
+def test_user_client(api_client, test_user):
```
Comment (matthewelwell) [inline api/conftest.py]: I'm not sure why we've had to create new fixtures here? We already have the `admin_client` fixture don't we?

File: api/tests/unit/edge_api/identities/test_edge_identity_featurestate_view.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+from django.urls import reverse
+from rest_framework import status
+
+
+def test_user_with_manage_identities_permission_can_retrieve_all_feature_states_for_identity(
```
Comment (matthewelwell) [inline api/tests/unit/edge_api/identities/test_edge_identity_featurestate_view.py]: seems like it's the addition of the VIEW_ENVIRONMENT permission that's actually the one that's providing access here. Perhaps that's correct though?

File: api/tests/unit/environments/identities/test_identities_feature_states_views.py [hunk 3]
```diff
@@ -84,3 +84,32 @@ def test_user_with_update_feature_state_permission_can_update_identity_feature_s
 
     # Then
     assert response.status_code == status.HTTP_201_CREATED
+
+
+def test_user_with_manage_identities_permission_can_retrieve_all_feature_states_for_identity(
```
Comment (matthewelwell) [inline api/tests/unit/environments/identities/test_identities_feature_states_views.py]: As above, I think it's the VIEW_ENVIRONMENT permission that's doing the work here?

File: api/conftest.py (hunk 1 above)
Comment (gagantrivedi) [reply api/conftest.py]: Yeah, but that uses the admin user (admin user have access to everything, right? )

File: api/tests/unit/edge_api/identities/test_edge_identity_featurestate_view.py (hunk 2 above)
Comment (gagantrivedi) [reply api/tests/unit/edge_api/identities/test_edge_identity_featurestate_view.py]: Ah yes, I have updated the test name

File: api/conftest.py (hunk 1 above)
Comment (matthewelwell) [reply api/conftest.py]: I don't think so, no. They are a django super user (I think) but that doesn't give them access to any specific organisations.

Comment (gagantrivedi) [reply api/conftest.py]: No, the admin user is actually org admin
https://github.com/Flagsmith/flagsmith/blob/main/api/conftest.py#L33

Comment (matthewelwell) [reply api/conftest.py]: Ah yes, if you include the organisation (which most fixtures do eventually I suppose). Ok, then can we try and organise the fixtures a bit better at least so it's (hopefully) a bit more obvious why we have 2 client fixtures? i.e. let's put the user fixture next to the admin_user fixture and the client fixture next to the admin_client fixture.

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**